### PR TITLE
TransposeSinking transformation: support new ops and fix performance issues

### PR DESCRIPTION
**Pull request by itikhono**
- All new **TransposeSinkingGeneral** transformations were moved to separate files and are used in TF Frontend only in this Release.
- **Legacy TransposeSinking** transformation is not affected by these changes and is used in MOC/Common optimization pipeline.
- **OVTF TransposeSinking** transformation is not not used in our pipelines and planned to be removed in the next PR.

Changes:

- Supported new ops in TransposeSinkingGeneral transformation: BatchToSpace, SpaceToBatch, Interpolate, Unsqueeze, VariadicSplit, Reduction backward and some other Unary/Binary ops.
- Updated transformations:
  TransposeSinkingFuse transformation to cover the case with multiple outputs. 
  TransposeSinking for Squeeze to support when 2nd input to Squeeze is empty.
- Updated tests: added new tests with the common template for all operations. Covered new supported operations.
- Bug fix.

### Tickets:
 - *103398, 103654, 101756, 88277, 103419, 103417, 100920, 100921, 102637*

### CODE REVIEW

**Review by rkazants — COMMENTED** [review]
please also enable test https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/tests/convert_tricky_models.cpp#L226

**Comment by itikhono** [comment]
> please also enable test https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/tests/convert_tricky_models.cpp#L226

The test has been enabled

**Review by evkotov — APPROVED** [review]

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking_interpolate.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            return false;
+        }
+
+        auto transpose_const = as_type_ptr<Constant>(pattern_to_node.at(const_label));
+        if (!transpose_const) {
+            return false;
+        }
+
+        // remove Transpose on 1st input:
+        auto transpose_parent = main_node->input_value(0).get_node()->input_value(0);
```
Comment (rkazants) [inline src/common/transformations/src/transformations/common_optimizations/transpose_sinking_interpolate.cpp:43]: can it be a case that `main_node->input_value(0).get_node()` is a node that does not have any input like `Parameter` or `Const`

**Review by rkazants — COMMENTED** [review]
do we have test for Interpolate?

**Review by rkazants — APPROVED** [review]

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp [hunk 2]
```diff
@@ -26,9 +26,15 @@ std::shared_ptr<opset6::Constant> get_reduced_order_constant(const std::shared_p
     auto order = order_const->cast_vector<int64_t>();
 
     auto axes = axes_const->cast_vector<int64_t>();
-    std::sort(axes.rbegin(), axes.rend());
-    for (const auto& i : axes)
-        order.erase(order.begin() + i);
+    if (!axes.empty()) {
+        std::sort(axes.rbegin(), axes.rend());
+        for (const auto& i : axes)
+            order.erase(order.begin() + i);
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp:32]: this is not a part of these changes, 

> do we have test for Interpolate?

Yes, we do, I added new common tests for many ops including Interpolate (transpose_sinking_common_tests.cpp)

**Comment by itikhono** [comment]
We agreed to resolve these comments in the next PR

**Comment by itikhono** [comment]
> Reference

> do we have test for Interpolate?

Yes, we do, I've added new common tests for many ops including Interpolate (transpose_sinking_common_tests.cpp)

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking_interpolate.cpp (hunk 1 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/common_optimizations/transpose_sinking_interpolate.cpp:43]: this is Transpose node,  fixed https://github.com/openvinotoolkit/openvino/pull/16283

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking_fuse.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                ov::replace_output_update_name(out_transpose.get_node()->output(0), input);
+            }
+        } else {
+            auto new_order = Constant::create(transpose_order_type, {order2.size()}, order2);
+            auto new_transpose = register_new_node<Transpose>(input, new_order);
+
+            new_transpose->set_friendly_name(m.get_match_root()->get_friendly_name());
+            transpose_sinking::RemoveSingleOutputConsumers(transpose1);
+            copy_runtime_info(transpose1, new_transpose);
+            ngraph::replace_node(transpose1, new_transpose);
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/src/transformations/common_optimizations/transpose_sinking_fuse.cpp:64]: fixed https://github.com/openvinotoolkit/openvino/pull/16283

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking_fuse.cpp [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            return false;
+
+        auto order1 = transpose1_order->cast_vector<int64_t>();
+        auto order2 = transpose2_order->cast_vector<int64_t>();
+        if (order1.size() != order2.size())
+            return false;
+
+        bool is_ordered = true;
+        for (size_t i = 0; i < order1.size(); i++) {
+            order2[i] = order1[order2[i]];
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/src/transformations/common_optimizations/transpose_sinking_fuse.cpp:44]: fixed https://github.com/openvinotoolkit/openvino/pull/16283

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking_concat.cpp [hunk 5]
```diff
@@ -43,15 +39,17 @@ ov::pass::TransposeSinkingConcatForward::TransposeSinkingConcatForward() {
             return false;
         }
 
+        const auto transpose_axis_order = transpose_input_info.transpose_const->get_axis_vector_val();
+        const int64_t transposed_concat_axis = transpose_axis_order[concat_axis];
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/src/transformations/common_optimizations/transpose_sinking_concat.cpp:43]: fixed https://github.com/openvinotoolkit/openvino/pull/16283

File: src/common/transformations/include/transformations/common_optimizations/transpose_sinking_utils.hpp [hunk 6]
```diff
@@ -10,6 +10,7 @@
 
 #include "openvino/op/util/op_types.hpp"
 #include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/graph_rewrite.hpp"
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/include/transformations/common_optimizations/transpose_sinking_utils.hpp:13]: fixed https://github.com/openvinotoolkit/openvino/pull/16283

File: src/common/transformations/include/transformations/common_optimizations/transpose_sinking_data_movement.hpp [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace ov {
+namespace pass {
+
+class TRANSFORMATIONS_API TransposeSinkingDataMovementForward;
+class TRANSFORMATIONS_API TransposeSinkingDataMovementBackward;
+
+}  // namespace pass
+}  // namespace ov
+
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/include/transformations/common_optimizations/transpose_sinking_data_movement.hpp:19]: done https://github.com/openvinotoolkit/openvino/pull/16283

File: src/common/transformations/include/transformations/common_optimizations/transpose_sinking_data_movement.hpp [hunk 8]
```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2022-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/include/transformations/common_optimizations/transpose_sinking_data_movement.hpp:7]: fixed https://github.com/openvinotoolkit/openvino/pull/16283

**Pull request merged**